### ISSUE TITLE
frontend does not respect security (`https`) of base URL

### ISSUE THREAD

**Issue by damian0815** [invoke-ai/InvokeAI#973]
I'm trying to run on a runpod.io instance, but the frontend's socket communications doesn't seem to be respecting the security/protocol of the base URL:

```
Mixed Content: The page at 'https://<uuid>-<port>.proxy.runpod.io/' was loaded over HTTPS, but requested 
an insecure XMLHttpRequest endpoint 
'http://<uuid>-<port>.proxy.runpod.io/socket.io/?EIO=4&transport=polling&t=OEoR1LP'. 
This request has been blocked; the content must be served over HTTPS.
```

(This is running against current `development` branch).

The workaround is to do SSH tunnelling, but you pay more on runpod for an instance that allows that.

**Comment by damian0815** [comment]
I think I fixed it, please confirm in the PR

**Comment by lstein** [comment]
I believe this is now fixed. Please confirm.

**Comment by simpleyuji** [comment]
im seeing this error as well on commit 194d4c75b39b09a48d5facdeb1b1a7e5de972203

**Comment by psychedelicious** [comment]
Resolved by #976.
